src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1198,6 +1216,8 @@ public void RemoveFromWorkspace()
                     throw new InvalidOperationException("The project has already been removed.");
                 }
 
+                _asynchronousFileChangeProcessingCancellationTokenSource.Cancel();
```

Comment:
ah yeah, i guesss we can stop the queue when the project is removed.  alternatively, don't bother with this special CTS and instead use hte shutdown token, and just have the processing loop check if the project has been removed.